### ISSUE TITLE
Make rounded corners for rectangles

### ISSUE THREAD

**Issue by StaffEngineer** [StaffEngineer/velo#21]
Let's make pretty UI with rounded corners for rectangles, buttons, etc.

(possibly https://github.com/bevyengine/bevy/pull/3991 have to be released first)

**Comment by StaffEngineer** [comment]
> why not try other bevy ui plugin like [bevy_egui](https://github.com/mvlabat/bevy_egui/) or [belly](https://github.com/jkb0o/belly)

When I first started using Rust, I looked into using egui to add a handler for a whole table row click. However, I wasn't satisfied with the API. While the belly library is interesting, it doesn't seem to offer much for this particular project. Additionally, I don't believe that belly supports rounded corners.

While belly may not offer a solution for all of the requirements of this project, I am still open to considering it if it can bring something useful to the table that can't be done easily with raw bevy-ui. That being said, the main limitation of using raw bevy-ui for this project is not so much the lack of rounded corners, but rather the absence of editable/selectable text input.

I've spoken with bevy-ui contributors on Discord, and they have informed me that migration to cosmic-text will enable editable text input for bevy-ui. The migration is expected to occur in the upcoming 0.11 release. My goal for this project is to have rich text editing with support for markdown/org syntax in rectangles, and potentially even executable code (literate programming) in the future.
